core/services/gateway/handlers/webapicapabilities/handler.go

```diff
@@ -1,6 +1,116 @@
 package webapicapabilities
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/config"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers"
+)
+
 const (
 	// NOTE: more methods will go here. HTTP trigger/action/target; etc.
-	MethodWebAPITarget = "web_api_target"
+	MethodWebAPITarget  = "web_api_target"
+	MethodWebAPITrigger = "web_api_trigger"
 )
+
+type handler struct {
+	config         HandlerConfig
+	donConfig      *config.DONConfig
+	don            handlers.DON
+	savedCallbacks map[string]*savedCallback
+	mu             sync.Mutex
+	lggr           logger.Logger
+}
+
+type HandlerConfig struct {
+	MaxAllowedMessageAgeSec uint
+}
+type savedCallback struct {
+	id         string
+	callbackCh chan<- handlers.UserCallbackPayload
+}
+
+var _ handlers.Handler = (*handler)(nil)
+
+func NewWorkflowHandler(handlerConfig json.RawMessage, donConfig *config.DONConfig, don handlers.DON, lggr logger.Logger) (*handler, error) {
+	var cfg HandlerConfig
+	err := json.Unmarshal(handlerConfig, &cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &handler{
+		config:         cfg,
+		donConfig:      donConfig,
+		don:            don,
+		savedCallbacks: make(map[string]*savedCallback),
+		lggr:           lggr.Named("WorkflowHandler." + donConfig.DonId),
+	}, nil
+}
+
+func (d *handler) HandleUserMessage(ctx context.Context, msg *api.Message, callbackCh chan<- handlers.UserCallbackPayload) error {
+	d.mu.Lock()
+	d.savedCallbacks[msg.Body.MessageId] = &savedCallback{msg.Body.MessageId, callbackCh}
+	don := d.don
+	d.mu.Unlock()
+	body := msg.Body
+	var payload TriggerRequestPayload
+	err := json.Unmarshal(body.Payload, &payload)
```

Comment:
This fills empty fields with default values. Since we don't do any validation on the values, what would go wrong if an empty request is sent?

Probably worth a test

Reply:
Good idea.  I've also added a test for an invalid payload.